### PR TITLE
added eslint rules for allSettledPromises

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: https://issues.redhat.com/browse/RHOAIENG-1295

## Description
added eslint to restrict using Promise.allSetttled

## How Has This Been Tested?
npm run test

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/utilities/allSettledPromises.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ type TypedPromiseRejectedResult<R> = Omit<PromiseRejectedResult, 'reason'> & { r
 export const allSettledPromises = <T, E = undefined>(
   data: Promise<T>[],
 ): Promise<[PromiseFulfilledResult<T>[], TypedPromiseRejectedResult<E>[]]> =>
+  // eslint-disable-next-line no-restricted-properties
```
Comment (manaswinidas) [inline frontend/src/utilities/allSettledPromises.ts:6]: Any specific reasons why some were disabled and some were replaced?

**Review by manaswinidas — CHANGES_REQUESTED** [review]

File: frontend/src/concepts/pipelines/content/DeletePipelineRunsModal.tsx [hunk 2]
```diff
@@ -66,6 +66,7 @@ const DeletePipelineRunsModal: React.FC<DeletePipelineRunsModalProps> = ({
               setDeleting(false);
             });
         } else {
+          // eslint-disable-next-line no-restricted-properties
```
Comment (christianvogt) [inline frontend/src/concepts/pipelines/content/DeletePipelineRunsModal.tsx:69]: Please add a comment when disabling any rule. In this particular case it was discussed that Andrew would prefer a refactor for the use case to not depend on the index in the delete modal.

File: frontend/.eslintrc [hunk 3]
```diff
@@ -178,6 +178,14 @@
         ]
       }
     ],
+    "no-restricted-properties": [
+      "error",
+      {
+        "object": "Promise",
+        "property": "allSettled",
+        "message": "Avoid using Promise.allSettled use allSettledPromises utility function instead"
```
Comment (christianvogt) [inline frontend/.eslintrc:186]:
```suggestion
        "message": "Avoid using Promise.allSettled, use allSettledPromises utility function instead."
```

File: frontend/src/utilities/allSettledPromises.ts (hunk 1 above)
Comment (christianvogt) [inline frontend/src/utilities/allSettledPromises.ts:6]:
```suggestion
  // Use of `Promise.allSettled` is justified as this utility is a wrapper with type improvements.
  // eslint-disable-next-line no-restricted-properties
```

**Pull request merged**